core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4562,11 +4616,10 @@ public void run() {
                 try (ACLContext ctx = ACL.as2(ACL.SYSTEM2)) {
 
                     // Wait 'til we have no active executors.
-                    doQuietDown(true, 0, null);
-
+                    doQuietDown2(true, 0, message, true);
                     // Make sure isQuietingDown is still true.
                     if (isQuietingDown()) {
-                        servletContext.setAttribute("app", new HudsonIsRestarting());
+                        servletContext.setAttribute("app", new JenkinsIsSafeRestarting());
```

Comment:
To clarify, the idea here is to show a different message on "restarting" (unsafely) and during safe-restart.

Would it not safe a lot of duplicated code by passing in a `boolean` flag to `HudsonIsRestarting` (or even the system message)?

Reply:
This has been implemented, good idea